e2e/testcases/csr_auth_test.go

```diff
@@ -162,64 +162,62 @@ func TestFleetWIDifferentProject(t *testing.T) {
 	})
 }
 
-// getMembershipIdentityProvider fetches the workload_identity_pool if the membership exists.
-func getMembershipIdentityProvider(nt *nomostest.NT) (bool, string, error) {
-	bytes, err := nt.Kubectl("get", "membership", "membership")
+// getMembership gets the membership resource in the GCP project.
+// It returns a boolean that indicates whether the membership exists, and an error
+func getMembership(fleetMembership, gcpProject string) (bool, error) {
+	bytes, err := exec.Command("gcloud", "container", "fleet", "memberships", "describe", fleetMembership, "--project", gcpProject).CombinedOutput()
 	out := string(bytes)
 	if err != nil {
-		if strings.Contains(out, `the server doesn't have a resource type "membership"`) || strings.Contains(out, "NotFound") {
-			return false, "", nil
+		// ERROR: (gcloud.container.fleet.memberships.describe) Membership xxx not found in the fleet.
+		if strings.Contains(out, "not found") {
+			return false, nil
 		}
-		return false, "", fmt.Errorf("unable to get the membership %s: %w", out, err)
+		return false, fmt.Errorf("failed to describe the membership %s in project %s (out: %s\nerror: %w)", fleetMembership, gcpProject, out, err)
 	}
 
-	bytes, err = nt.Kubectl("get", "membership", "membership", "-o", "jsonpath={.spec.workload_identity_pool}")
-	out = string(bytes)
-	if err != nil {
-		return true, "", fmt.Errorf("unable to get the membership workload identity %s: %w", out, err)
-	}
-	return true, out, nil
+	return true, nil
 }
 
 // cleanMembershipInfo deletes the membership by unregistering the cluster.
 // It also deletes the reconciler-manager to ensure the membership watch is not set up.
 func cleanMembershipInfo(nt *nomostest.NT, fleetMembership, gcpProject, gkeURI string) {
-	membershipExists, wiPool, err := getMembershipIdentityProvider(nt)
+	exists, err := getMembership(fleetMembership, gcpProject)
 	if err != nil {
 		nt.T.Error(err)
-		nt.T.Skip("Skip the test because unable to check if membership exists")
-	} else if membershipExists {
-		fleetProject := gcpProject
-		if len(wiPool) > 0 {
-			fleetProject = strings.TrimSuffix(wiPool, ".svc.id.goog")
+		nt.T.Skip("Skip the test because unable to check get membership and its workload identity pool")
```

Comment:
I suspect this should just Fatal, not Skip.

Reply:
DONE